### PR TITLE
Bump LLVM

### PR DESCRIPTION
**Pull request by chelini**

### CODE REVIEW

**Comment by chelini** [comment]
I had to disable one test: Integration/transform/transform-convolutions.mlir, unclear why it is failing.

**Comment by adam-smnk** [comment]
I'll test GPU stuff locally.

**Review by rengolin — COMMENTED** [review]
would be interesting to know if the new dealloc logic makes some of our tests not need to disable asan.

**Comment by adam-smnk** [comment]
CUDA code generation needs to get adjusted and its GPU tests updated.
Vulkan crashes at runtime. I'll investigate further.

**Comment by adam-smnk** [comment]
All test now pass both locally and on the new cluster nodes.
The bump should be ready to land.

File: lib/TPP/Bufferize.cpp [hunk 1]
```diff
@@ -99,13 +102,21 @@ void Bufferize::runOnOperation() {
...
     passManager.addNestedPass<func::FuncOp>(createCanonicalizerPass());
   }
+  passManager.addPass(bufferization::createDropEquivalentBufferResultsPass());
+
+  if (dealloc) {
+    bufferization::BufferDeallocationPipelineOptions options;
+    bufferization::buildBufferDeallocationPipeline(passManager, options);
+  }
 
+  passManager.addPass(createBufferizationToMemRefPass());
```
Comment (adam-smnk) [inline lib/TPP/Bufferize.cpp:112]: Is this needed together with oneshot?
If yes, then shouldn't it be placed before the deallocation pass?

**Review by adam-smnk — APPROVED** [review]
LGTM
Automatic deallocation is disabled for CUDA as unified memory is not supported everywhere. However, it works well with Vulkan so, there the deallocation is enabled.
GPU pipeline has been updated to align with upstream cubin binary generation changes.
@rengolin you should make one more pass just to double check us

File: lib/TPP/Bufferize.cpp (hunk 1 above)
Comment (chelini) [reply lib/TPP/Bufferize.cpp:112]: is needed to convert bufferization operations to memref (i.e., bufferization.dealloc or bufferization.clone).

File: test/Integration/transform/transform-convolutions.mlir [hunk 2]
```diff
@@ -10,6 +10,8 @@
 // RUN:  -e entry -entry-point-result=void | \
 // RUN: FileCheck %s
 
+// XFAIL: *
```
[the remark this answers is not in the document]
Comment (chelini) [reply test/Integration/transform/transform-convolutions.mlir:13]: error: memory free side-effect on MemRef value not supported!

[the remark this answers is not in the document]
Comment (chelini) [reply test/Integration/transform/transform-convolutions.mlir:13]: It comes from the buffer deallocation pass, I still need to understand what is going on.

**Review by rengolin — APPROVED** [review]
Should be fine, it's mostly a lot of mechanical changes for maxf and the deallocation.

**Pull request merged**